src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            List<Bulldozer> machines = new BulldozerProducer().get();
+            return new ArrayList<>(machines);
```

Comment:
```suggestion
        if (type == Bulldozer.class) {
            return new ArrayList<>(bulldozerProducer.get());
```
lets create bulldozers as class-variables - to avoid creating a new producer each time this method is invoked
remember about oop principles